### PR TITLE
Update dependencies and refactor schema handling

### PR DESCRIPTION
**Pull request by koriym**
### Description

- Drop PHP 7.x and 8.0 support

This pull request includes multiple updates and refactors to improve the project:
- Upgraded PHPUnit to version `^9.6.22` for compatibility with the latest features and fixes.
- Replaced the `Json` class with `IniJson` for better naming consistency and improved error handling for INI parsing.
- Refactored JSON schema generation to remove external library dependency (`php-json-schema-generator`), opting for manual schema creation to streamline the codebase.
- Introduced custom exceptions (`InvalidEnvJsonFormatException`, `InvalidIniFileException`, and others) to enhance error reporting and granularity.
- Improved exception handling, reducing redundancy and increasing clarity across multiple classes.
- Restructured JSON decoding logic for improved readability and maintainability.
- Adjusted tests to align with updated functionality and new exception cases.

### Checklist

- [ ] Tests have been updated to reflect changes.
- [ ] Documentation has been reviewed and updated where necessary.
- [ ] No unnecessary dependencies remain in the project. 

Please review these changes and provide feedback as necessary.

## Sourcery によるサマリー

EnvJson ライブラリをリファクタリングし、エラー処理の強化、依存関係の更新、コード品質の向上を行いました。

新機能:
- JSON ファイルから環境変数をロードするためのコマンドラインツールを導入し、複数の出力形式に対応しました。
- カスタム例外による、よりきめ細かいエラー処理のサポートを追加しました。

バグ修正:
- 環境変数のパースとバリデーションに関する問題を修正しました。
- JSON ファイルの欠落または無効な場合の処理を改善しました。

機能拡張:
- JSON スキーマ生成をリファクタリングし、外部ライブラリへの依存関係を削除しました。
- JSON および環境変数処理ロジックを改善しました。
- エラー報告と例外処理を強化しました。
- コードを再構成し、可読性と保守性を向上させました。

ビルド:
- PHP のバージョン要件を 8.1 以上に更新しました。
- 開発用依存関係を更新しました。
- 開発ツールを管理するために bamarni/composer-bin-plugin を追加しました。

ドキュメント:
- README を更新し、新しいコマンドラインツールの使用方法を記載しました。
- インラインドキュメントとコードコメントを改善しました。

テスト:
- さまざまなエッジケースに対するテストカバレッジを拡張しました。
- 新しいコマンドラインツールの包括的なテストを追加しました。
- 新しいエラー処理シナリオをカバーするようにテストスイートを改善しました。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Refactor and improve the EnvJson library with enhanced error handling, dependency updates, and code quality improvements

New Features:
- Introduced a command-line tool for loading environment variables from JSON files with multiple output formats
- Added support for more granular error handling with custom exceptions

Bug Fixes:
- Fixed issues with environment variable parsing and validation
- Improved handling of missing or invalid JSON files

Enhancements:
- Refactored JSON schema generation to remove external library dependency
- Improved JSON and environment variable handling logic
- Enhanced error reporting and exception handling
- Restructured code for better readability and maintainability

Build:
- Updated PHP version requirement to 8.1+
- Updated development dependencies
- Added bamarni/composer-bin-plugin for managing development tools

Documentation:
- Updated README with new command-line tool usage instructions
- Improved inline documentation and code comments

Tests:
- Expanded test coverage for various edge cases
- Added comprehensive tests for new command-line tool
- Improved test suite to cover new error handling scenarios

</details>

## Summary by CodeRabbit

- **New Features**
  - Introduced a command-line utility for loading and converting environment configurations in multiple formats.
  - Added new JSON schema and configuration files to enhance environment variable management.
  
- **Documentation**
  - Expanded user guides with detailed installation instructions, usage examples, and best practice sections.
  
- **Refactor**
  - Streamlined environment loading processes with enhanced error handling and simplified configuration checks.
  
- **Tests**
  - Added a comprehensive suite to verify correct behavior and error handling for environment variable processing.
  
- **Chores**
  - Updated automation workflows and dependency configurations to support modern PHP versions.

### CODE REVIEW

File: tests/EnvJsonTest.php [hunk 1]
```diff
@@ -4,56 +4,347 @@
...
+            $schemaFile = $testDir . '/env.schema.json';
+            if (file_exists($envFile)) {
+                unlink($envFile); // Uncomment if file is dynamically created
+            }
+
+            if (file_exists($schemaFile)) {
+                unlink($schemaFile); // Uncomment if file is dynamically created
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (koriym) [reply tests/EnvJsonTest.php]: Already fixed.

[the remark this answers is not in the document]
Comment (koriym) [reply tests/EnvJsonTest.php]: Unnecessary test. Removed.
